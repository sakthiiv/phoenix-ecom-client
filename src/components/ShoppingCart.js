import React, { Component } from "react";
import cart from "../api/cart";
import { Link } from "react-router-dom";
import ProductItem from "./ProductItem";
import sessionstorage from "../helpers/sessionstorage";
class ShoppingCart extends React.Component {
  state = {
    products: []
  };
  handleChange = e => {
      this.setState({ [e.target.name]: e.target.value });
  };

  async componentDidMount() {
    const  user  = sessionstorage.getItemFromSession('user');
    if (user && user.userId) {
      await this.fetchCartList(user.userId);
    }
  }

  fetchCartList = userId => {
    cart
     .cartGetDetails(userId)
     .then(responseJson => {
       this.setState({ products: responseJson.products });
     })
     .catch(error => {
       console.log(error);
     });
  }

  render() {
    let { products } = this.state;
    let prodList = products.map((product, i) => (
        <div className="col-sm-4"><ProductItem key={product.id} productItem={product} showDelete='true'/></div>
    ));
    return (
      <div>
            <h1 style={{marginLeft:"20px"}}>Shopping cart</h1>
            <div className="col-md-12 plist_container" style={{marginTop:"40px"}}>
                {prodList.length !== 0 ? <div>
                  <Link to={{ pathname: '/checkout', state: { products: products} }} className="btn btn-primary" style={{marginBottom:"30px", float:"right",marginRight:"50px"}}>
                    Proceed to checkout
                  </Link>
                  <div className="clearfix plist_container" style={{float:"left"}}>{prodList}</div>
                </div>: <h2 style={{color:"red"}} >No products in the cart</h2>}
            </div>
            
      </div>
    );
  }
}

export default ShoppingCart;