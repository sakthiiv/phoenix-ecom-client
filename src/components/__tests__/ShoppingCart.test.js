import React from 'react';
import { shallow } from 'enzyme';
import ShoppingCart from '../ShoppingCart';
describe('Admin', () => {
    it('renders correctly', () => {
      const wrapper = shallow(<ShoppingCart />);
      expect(wrapper).toMatchSnapshot();
      // On the first run of this test, Jest will generate a snapshot file automatically.
    });
});