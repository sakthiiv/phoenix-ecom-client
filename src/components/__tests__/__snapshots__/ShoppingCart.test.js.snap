// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Admin renders correctly 1`] = `
<div>
  <h1
    style={
      Object {
        "marginLeft": "20px",
      }
    }
  >
    Shopping cart
  </h1>
  <div
    className="col-md-12 plist_container"
    style={
      Object {
        "marginTop": "40px",
      }
    }
  >
    <h2
      style={
        Object {
          "color": "red",
        }
      }
    >
      No products in the cart
    </h2>
  </div>
</div>
`;
